### PR TITLE
Core, BigQuery: refactor 'client_info' support.

### PR DESCRIPTION
**Pull request by tseaver**
Hoist `client_info` support from `google.cloud.bigquery._http.Connection` into core's `google.cloud._http.Connection`.

Toward #7825.

### CODE REVIEW

**Comment by tseaver** [comment]
@tswast 
> We'll need to bump the min version of the dependency on google-cloud-core in google-cloud-bigquery, though, right?

Yes, and for the other manual / JSON clients which will rely on this feature in future PRs.

File: api_core/google/api_core/client_info.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+_PY_VERSION = platform.python_version()
+_API_CORE_VERSION = pkg_resources.get_distribution("google-api-core").version
+
+try:
+    _GRPC_VERSION = pkg_resources.get_distribution("grpcio").version
+except pkg_resources.DistributionNotFound:  # pragma: NO COVER
+    _GRPC_VERSION = None
+
+
+class ClientInfo(object):
```
Comment (tseaver) [inline api_core/google/api_core/client_info.py]: @tswast In order to support the API libraries which do not install grpc, I had to split out a base `ClientInfo` class which is importable if 'grpc' is absent.  The one in `google.api_core.gapic_v1.client_info` subclasses and adds the `to_grpc_metadata` method.

Comment (tswast) [reply api_core/google/api_core/client_info.py]: Thanks! Good catch.

File: api_core/google/api_core/client_info.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import platform
+
+import pkg_resources
+
+_PY_VERSION = platform.python_version()
+_API_CORE_VERSION = pkg_resources.get_distribution("google-api-core").version
+
+try:
+    _GRPC_VERSION = pkg_resources.get_distribution("grpcio").version
+except pkg_resources.DistributionNotFound:  # pragma: NO COVER
```
Comment (tswast) [inline api_core/google/api_core/client_info.py]: Should we add a unit test for this? It'd be good to be doubly sure that it doesn't blow up when `_GRPC_VERSION` is `None`.

**Comment by tswast** [comment]
Let's add `google.api_core.client_info` to the docs, similar to https://github.com/googleapis/google-cloud-python/blob/a017937e307fa437d1bf5fb682bb29a9c16f0a67/docs/core/page_iterator.rst

**Comment by cgparkinson** [comment]
Probably the fault of my setup, but a nightly test failed last night with the error
`ImportError: No module named client_info`

This error has no results on Google so thought I'd post here.

I fixed the error by adding `google-cloud-core<1.0.0` to my dependencies.

Probably will be better to update all my dependencies rather than relying on old versions though.

In case it's useful, my dependencies now read:

```
pandas-gbq
google-cloud-bigquery>=1.9.0
google-cloud-core<1.0.0
google-cloud-storage
parmap
google-api-python-client
google-cloud-pubsub>=0.28.1
```
